### PR TITLE
docs: updated website documentations to reflect current features

### LINKED ISSUE #1071: Improve Penrose style language documentation
Many feature of Penrose's Style language are not documented anywhere. We should write documentation for all of these features and add them to the website under the [Style header](https://penrose.cs.cmu.edu/docs/ref/style/).

### PR DESCRIPTION
**Pull request by liangyiliang**
# Description

Fixes #1071 

I updated the documentation about syntax and semantics on the Penrose website to better reflect the features of the current version of Penrose.

I also changed the order of the documentation in the sidebar from `Style, Substance, Domain` to `Domain, Substance, Style` so that later chapters depend on prior chapters.

## Domain
- Type Declaration (Subtypes)
- Predicate Declaration (`Prop` type; symmetric predicates)
- Function / Constructor Declarations
- Comments

## Substance
- Object Declarations (Substance variables)
- Predicate Applications
- Function / Constructor Applications
- Comments

## Style
- Namespaces
- Style Blocks
- - Matching semantics
- - Symmetric relations
- - Special `object has field` relations

### CODE REVIEW

File: packages/docs-site/docs/ref/domain/syntax.mdx [hunk 1]
```diff
@@ -5,4 +5,142 @@ pagination_next: null
...
+```
+type FirstType, SecondType
+
+predicate P1 (FirstType, SecondType)
+predicate P2 (SecondType a1, SecondType a2)
+predicate P3 ()
+predicate P4 (FirstType, SecondType, SecondType)
+```
+
+Notice how some declarations have names associated with the argument types (e.g. `P2` with argument names `a1` and `a2`), and some don't. These names to argument types are allowd and encouraged to enhance readability, but are not required.
```
Comment (samestep) [inline packages/docs-site/docs/ref/domain/syntax.mdx:80]:
```suggestion
Notice how some declarations have names associated with the argument types (e.g. `P2` with argument names `a1` and `a2`), and some don't. These names to argument types are allowed and encouraged to enhance readability, but are not required.
```

File: packages/docs-site/docs/ref/domain/syntax.mdx [hunk 2]
```diff
@@ -5,4 +5,142 @@ pagination_next: null
...
+
+### The `Prop` Type
+
+A special type, _Prop_, is built into _Penrose_, and is available to be used as predicate arguments. The _Prop_ type represents predicates themselves. For example, the predicate _Not_ is declared in the set-theory example:
```
Comment (samestep) [inline packages/docs-site/docs/ref/domain/syntax.mdx:84]: Do we want to italicize _Penrose_ everywhere it appears?

File: packages/docs-site/docs/ref/domain/syntax.mdx [hunk 3]
```diff
@@ -5,4 +5,142 @@ pagination_next: null
...
+
+```
+predicate Not (Prop p1)
+```
+
+When we use the _Not_ predicate in the _substance_ or _style_ schemas, then, we can pass in any predicates as _Prop_. For example, we can write `Not(Intersection(s1, s2))` in the _substance_ or _style_ schemas to conceptually denote that `s1` and `s2` are disjoint.
```
Comment (samestep) [inline packages/docs-site/docs/ref/domain/syntax.mdx:90]: Perhaps this reads a bit more clearly?
```suggestion
When we use the _Not_ predicate in the _substance_ or _style_ schemas, then, we can pass in any predicates as _Prop_. For example, we can write `Not(Intersecting(s1, s2))` in the _substance_ or _style_ schemas to conceptually denote that `s1` and `s2` are disjoint.
```

File: packages/docs-site/docs/ref/domain/syntax.mdx (hunk 2 above)
Comment (liangyiliang) [reply packages/docs-site/docs/ref/domain/syntax.mdx:84]: I am fine with both, but I do want to be consistent

File: packages/docs-site/docs/ref/domain/syntax.mdx (hunk 1 above)
Comment (liangyiliang) [reply packages/docs-site/docs/ref/domain/syntax.mdx]: Fixed, thanks

File: packages/docs-site/docs/ref/domain/syntax.mdx (hunk 3 above)
Comment (liangyiliang) [reply packages/docs-site/docs/ref/domain/syntax.mdx]: Fixed, thanks

File: packages/docs-site/docs/ref/domain/syntax.mdx (hunk 2 above)
Comment (liangyiliang) [reply packages/docs-site/docs/ref/domain/syntax.mdx:84]: @joshsunshine Any ideas about whether "Penrose" should be italicized?

File: packages/docs-site/docs/ref/domain/syntax.mdx [hunk 4]
```diff
@@ -5,4 +5,142 @@ pagination_next: null
...
+
+The arguments in `argument_list` contains the types of objects that this predicate accepts, separated by commas. Here is a _domain_ schema that contains a couple of type declarations and four different, valid, predicate declarations:
+
+```
+type FirstType, SecondType
+
+predicate P1 (FirstType, SecondType)
+predicate P2 (SecondType a1, SecondType a2)
+predicate P3 ()
+predicate P4 (FirstType, SecondType, SecondType)
```
Comment (samestep) [inline packages/docs-site/docs/ref/domain/syntax.mdx:77]: Speaking of consistency: it looks like in some places we have a space between the predicate name and the open parenthesis for the argument list, and in other places we don't. Can we choose one and stick to that? My vote is for no space.

File: packages/docs-site/docs/ref/style/syntax.mdx [hunk 5]
```diff
@@ -3,6 +3,316 @@ pagination_prev: null
...
+    field_1 = expr_1
+    field_2 = expr_2
+    ...
+}
+```
+
+Values declared within a namespace can be read outside of the namespace using the "dot" operator:
+
+```
+namespace_name . field_name
```
Comment (samestep) [inline packages/docs-site/docs/ref/style/syntax.mdx:30]:
```suggestion
namespace_name.field_name
```

**Review by samestep — COMMENTED** [review]
Looks awesome other than the few comments I left; thanks so much for doing this!

File: packages/docs-site/docs/ref/domain/syntax.mdx (hunk 2 above)
Comment (wodeni) [reply packages/docs-site/docs/ref/domain/syntax.mdx:84]: We smallcap Penrose in academic documents, but I don't think it's really necessary for the language reference.

File: packages/docs-site/docs/ref/domain/syntax.mdx (hunk 4 above)
Comment (liangyiliang) [reply packages/docs-site/docs/ref/domain/syntax.mdx:77]: I feel like adding spaces allows readers to better understand the structure of the exprs

Comment (samestep) [reply packages/docs-site/docs/ref/domain/syntax.mdx:77]: OK! Yeah, I don't have much of a preference between the two choices; I just want to be consistent. @joshsunshine @wodeni thoughts?

Comment (wodeni) [reply packages/docs-site/docs/ref/domain/syntax.mdx:77]: Concurring with @samestep here on the formatting.

Comment (liangyiliang) [reply packages/docs-site/docs/ref/domain/syntax.mdx:77]: I can make it to always have spaces in the docs

File: packages/docs-site/docs/ref/domain/syntax.mdx (hunk 2 above)
Comment (liangyiliang) [reply packages/docs-site/docs/ref/domain/syntax.mdx]: I will not italicize any occurence of "Penrose"

**Review by samestep — APPROVED** [review]

**Pull request merged**